doc/whats_new/v0.20.rst

```diff
@@ -255,6 +255,10 @@ Metrics
   False and both inputs are sparse, will return a sparse matrix.
   :issue:`10999` by :user:`Taylor G Smith <tgsmith61591>`.
 
+- :func:`metrics.cluster.silhouette_score` and
+  :func:`metrics.cluster.silhouette_samples` are more memory efficient,
+  causing them to run faster. :issue:`11135` by `Joel Nothman`_.
```

Comment:
As previously discussed, it would run faster only assuming that swapping occurs, meaning that the full distance matrix takes more than RAM size but less than RAM + SWAP disk size (typical sizes [here](https://askubuntu.com/a/49138)) which is IMO a too strong assumption.

Reply:
Given the reported freezes...?

Reply:
It's definitely significantly faster in light of benchmarks below - but I think because of a better implementation, not just due to chunking.

Maybe "are more memory efficient, and faster"?